client/components/header.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import SideNav from './side-nav';
+// import parseRoute from './lib/parse-route';
+
+class Header extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = {
+      navOpen: false,
+      title: ''
+    };
+  }
+
+  componentDidMount() {
+    const location = window.location.hash;
+    if (location === '#profile') {
+      this.setState({
+        title: 'Profile'
+      });
+    }
+    if (location === '#guests') {
+      this.setState({
+        title: 'Guest List'
+      });
+    }
+  }
```

Comment:
rather than storing this in state, i recommend defining a `getTitle()` method that just returns the correct string based on the `window.location.hash`. `componentDidMount` only runs once, so unless your header is unmounted for some reason, it'll get stuck with the wrong title.